tests/rbd/test_scale_group_snap_clone.py

```diff
@@ -0,0 +1,241 @@
+            rc = create_group_and_verify(**group_create_kw)
+            if rc:
+                log.error(f"group {group} creation failed, test case fail")
+                return 1
+            else:
+                log.info("STAGE: group creation succeeded")
+
+            for image, image_config in pool_config.items():
+                log.info(f"All image details : {image} and {image_config}")
+
+                # Add all 100+ images to the group using a loop.
+                log.info(f"Adding image: {image} in group {group}")
+                add_image_group_kw = {
+                    "client": client,
+                    "pool": pool,
+                    "group": group,
+                    "image": image,
+                }
+                rc = add_image_to_group_and_verify(**add_image_group_kw)
+                if rc:
+                    log.error(f"Image {image} add failed, test case fail")
+                    return 1
+                else:
+                    log.info(f"STAGE: image {image} add succeeded")
+
+            # Take a snapshot of the group using the rbd group snap create command.
+            group_snap = kw.get("config", {}).get("group_snap", "group_snap_default")
+            log.info(f"Creating group snapshot: {group_snap}")
+            snap_create_kw = {
+                "client": client,
+                "pool": pool,
+                "group": group,
+                "snap": group_snap,
+            }
+            rc = create_snap_and_verify(**snap_create_kw)
+            if rc:
+                log.error(f"snap {group_snap} create failure")
+                return 1
+            else:
+                log.info(f"STAGE: snap {group_snap} creation with validation done")
+
+            # Verify the group snapshot exists using the rbd group snap ls command.
+            log.info(f"Verify group id for group: {group} snap: {group_snap}")
+            group_ls_kw = {
+                "client": client,
+                "pool": pool,
+                "group": group,
+                "format": "json",
+            }
+            (g_ls_out, _) = group_info(**group_ls_kw)
+            log.info(g_ls_out)
+            g_ls_out = json.loads(g_ls_out)
+            if g_ls_out["group_id"].isalnum():
+                log.info(f"Group id exist for group {group}")
+            else:
+                log.error(f"Group id does not exist for group {group}")
+                return 1
+
+            # Get Snap-id
+            log.info(
+                "Get snap-id of group snapshot to further create clone of image from group snapshot"
+            )
+            snap_group_info = {
+                "client": client,
+                "pool": pool,
+                "group": group,
+                "snap": group_snap,
+                "format": "json",
+            }
+            (snap_g_out, _) = group_snap_info(**snap_group_info)
+            log.info(snap_g_out)
+            snap_g_out = json.loads(snap_g_out)
```

Comment:
can we keep this varibale more meaningful like `snap_group_output`?

Reply:
Done